enhancements/scorecard-storage.md

```diff
@@ -0,0 +1,250 @@
+
+This feature enables a wider adoption of tests that can be executed within scorecard.
+
+### Goals
+
+Goals of this proposal include:
+* provide a writable volume for scorecard test developers
+* Collect test output when scorecard completes
+
+### Non-Goals
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+A scorecard user can specify for a given test that they want storage to be made available to that test.  This is specified within the scorecard configuration file as a storage spec.  For example, a user would specify they want storage for a test by specifying
+in the scorecard configuration file the `storage` spec as in the following example:
+
+```yaml
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+    storage: 
+      spec:
+        mountPath:
+          path: /test-output
+```
+
+Note, that if the mountPath is not specified, then a default mount path
+of /test-output would be used.
+
+#### Story 2
+Scorecard gathers any persisted volume data upon test completion and stores it in a local directory where scorecard is executed.  Users
+can specify a non-default local path for test output using a command flag as follows:
+```bash
+operator-sdk scorecard ./bundle --selector=suite=custom --test-output=/mytestoutput
+```
+
+#### Story 3
+A scorecard user can specify storage globally for any selected tests.
+The `storage` spec can be defined globally as in the following example:
+
+```yaml
+  storage: 
+    spec:
+      mountPath:
+        path: /test-output
+  tests:
+  - image: quay.io/custom-scorecard-example/example-test:latest
+    entrypoint:
+    - example-scorecard-test
+    - example-test
+    labels:
+      suite: example
+      test: example-test
+```
+
+This would mean that any selected test would assume the global storage
+settings.  If a test has its own unique test storage defined, then those
+settings would override the global storage settings.
+
+### Implementation Details/Notes/Constraints
+
+For each test that specifies storage, scorecard will interate
+this sequence for each test:
+ * scorecard adds a sidecar container into the test pod, the sidecar container would also mount the same emptyDir volume that is mounted into the test container
+ * upon test container completion and before removing the test pod, scorecard
+will extract (exec/tar) the test output contents from the sidecar container to the local host that scorecard is executing upon
+ * scorecard cleanup would cleanup the test Pod as normal, thereby also removing the added storage sidecar container
+
+A successful scorecard execution of tests that use this storage feature
+would include a local directory of the collected test output, with test output being divided into subdirectories based on suite and test names
+
+### Risks and Mitigations
```

Comment:
what happens in the event that scorecard dies/terminates/is killed before collecting the output and telling the sidecar to terminate?  Presumably the test pod lives forever?

Do we want to include an option for a timeout in the sidecar?

Reply:
added a note just now that the sidecar would use the wait-time value for timeouts in the event of abnormal scorecard exit.